types/config.d.ts

```diff
@@ -42,16 +42,16 @@ export interface IConfig {
     // Send back instrumentation data, if set to true
     instrument?: boolean;
 
-    // Inspect CSSRuleList for style elements and send CSSRules data instead of style's text children
-    cssRules?: boolean;
+    // Skip writing cookie to track the user anonymously
+    skipCookie?: boolean;
 
     // Pointer to the function which would be responsible for sending the data
     // If left unspecified, raw payloads will be uploaded to the uploadUrl endpoint
     uploadHandler?: UploadHandler;
 
     // Pointer to the function which will be responsible for giving Clarity
     // a dictionary of strings that the user wants logged in each Clarity payload
-    customInstrumentation?: () => { [key: string]: string; };
+    customInstrumentation?: (impressionId?: string, cid?: string, projectId?: string) => { [key: string]: string; };
```

Comment:
Are we restricting to a fixed set?

Reply:
Not sure I understand the comment; The parameters are all optional (defined with a "?"). So we don't have to enforce. 

---
In reply to: [243140487](https://github.com/Microsoft/clarity-js/pull/116#discussion_r243140487) [](ancestors = 243140487)

Reply:
Are our custom instrumentation limited to these specific properties listed here (impressionId, etc.)?

---
In reply to: [245074855](https://github.com/Microsoft/clarity-js/pull/116#discussion_r245074855) [](ancestors = 245074855,243140487)

Reply:
For now yes; although I get your point. Let me turn this into a properties object that can include more or less features going forward. 

---
In reply to: [245143218](https://github.com/Microsoft/clarity-js/pull/116#discussion_r245143218) [](ancestors = 245143218,245074855,243140487)